app/uk/gov/hmrc/tariffclassificationfrontend/connector/FileStoreConnector.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tariffclassificationfrontend.connector
+
+import com.google.inject.Inject
+import javax.inject.Singleton
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.bootstrap.http.HttpClient
+import uk.gov.hmrc.tariffclassificationfrontend.config.AppConfig
+import uk.gov.hmrc.tariffclassificationfrontend.models._
+import uk.gov.hmrc.tariffclassificationfrontend.models.response.FileMetadata
+import uk.gov.hmrc.tariffclassificationfrontend.utils.JsonFormatters.fileMetaDataFormat
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+@Singleton
+class FileStoreConnector @Inject()(appConfig: AppConfig, http: HttpClient) {
```

Comment:
two methods to test and maintain for doing basically the same thing.... DRY principle

Reply:
DRY principle is a **principle** and should be followed pragmatically. These endpoints are fulfilling two **different** purposes

Reply:
One endpoint always returns a `200`. The other returns a `200` or `404` (if it doesnt exist). They both have different roles and are important